src/anomalib/pre_processing/pre_processing.py

```diff
@@ -0,0 +1,179 @@
+        >>> val_transform = Compose([Resize((256, 256)), CenterCrop((224, 224)), ToTensor()])
+
+        >>> # Create PreProcessor with stage-specific transforms
+        >>> pre_processor = PreProcessor(
+        ...     train_transform=train_transform,
+        ...     val_transform=val_transform
+        ... )
+
+        >>> # Create PreProcessor with a single transform for all stages
+        >>> common_transform = Compose([Resize((224, 224)), ToTensor()])
+        >>> pre_processor_common = PreProcessor(transform=common_transform)
+
+        >>> # Use in a Lightning module
+        >>> class MyModel(LightningModule):
+        ...     def __init__(self):
+        ...         super().__init__()
+        ...         self.pre_processor = PreProcessor(...)
+        ...
+        ...     def configure_callbacks(self):
+        ...         return [self.pre_processor]
+        ...
+        ...     def training_step(self, batch, batch_idx):
+        ...         # The pre_processor will automatically apply the correct transform
+        ...         processed_batch = self.pre_processor(batch)
+        ...         # Rest of the training step
+    """
+
+    def __init__(
+        self,
+        train_transform: Transform | None = None,
+        val_transform: Transform | None = None,
+        test_transform: Transform | None = None,
+        transform: Transform | None = None,
+    ) -> None:
+        super().__init__()
+
+        if transform and any([train_transform, val_transform, test_transform]):
+            msg = (
+                "`transforms` cannot be used together with `train_transform`, `val_transform`, `test_transform`.\n"
+                "If you want to apply the same transform to the training, validation and test data, "
+                "use only `transforms`. \n"
+                "Otherwise, specify transforms for training, validation and test individually."
+            )
+            raise ValueError(msg)
+
+        self.train_transform = train_transform or transform
+        self.val_transform = val_transform or transform
+        self.test_transform = test_transform or transform
+        self.predict_transform = self.test_transform
+        self.export_transform = get_exportable_transform(self.test_transform)
+
+    def setup_datamodule_transforms(self, datamodule: "AnomalibDataModule") -> None:
+        """Set up datamodule transforms."""
+        # If PreProcessor has transforms, propagate them to datamodule
+        if any([self.train_transform, self.val_transform, self.test_transform]):
+            transforms = {
+                "train": self.train_transform,
+                "val": self.val_transform,
+                "test": self.test_transform,
+            }
+
+            for stage in ["fit", "validate", "test", "predict"]:
+                transform = get_stage_transform(stage, transforms)
+                if transform is not None:
+                    set_datamodule_stage_transform(datamodule, transform, stage)
```

Comment:
This would further simplify the method and we could get rid of the `get_stage_transform` function
```suggestion
            transforms = {
                "train": self.train_transform,
                "val": self.val_transform,
                "test": self.test_transform,
                "predict": self.test_transform,
            }

            for stage, transform in transforms.items():
                if transform is not None:
                    set_datamodule_stage_transform(datamodule, transform, stage)
```

Reply:
Refactored in 3eecd89afb83d6ae09182cc2bf77df3fcd2c581e